src/ChunkReader.php

```diff
@@ -65,14 +68,24 @@ public function read(WithChunkReading $import, Reader $reader, TemporaryFile $te
             }
         }
 
-        $jobs->push(new AfterImportJob($import, $reader));
+        $afterImportJob = new AfterImportJob($import, $reader);
+
+        if ($import instanceof ShouldQueueWithoutChain) {
+            $jobs->push($afterImportJob->delay($delayCleanup));
+
+            return $jobs->each(function ($job) use ($queue) {
+                dispatch($job->onQueue($queue));
+            });
+        }
 
         if ($import instanceof ShouldQueue) {
             return new PendingDispatch(
                 (new QueueImport($import))->chain($jobs->toArray())
             );
         }
 
+        $jobs->push($afterImportJob);
```

Comment:
Why is the after import job not added to the chain that's being dispatched in the previous 'if'? This won't throw that event for jobs that go to queue.